lib/ui/views/messenger/bloc/create_discussion_form.dart

```diff
@@ -1,39 +1,38 @@
 part of 'providers.dart';
 
-final _createDiscussionGroupFormProvider = NotifierProvider.autoDispose<
-    CreateDiscussionGroupFormNotifier, CreateDiscussionGroupFormState>(
+final _createDiscussionFormProvider = NotifierProvider.autoDispose<
+    CreateDiscussionFormNotifier, CreateDiscussionFormState>(
   () {
-    return CreateDiscussionGroupFormNotifier();
+    return CreateDiscussionFormNotifier();
   },
 );
 
 @freezed
-class CreateDiscussionGroupFormState with _$CreateDiscussionGroupFormState {
-  const factory CreateDiscussionGroupFormState({
+class CreateDiscussionFormState with _$CreateDiscussionFormState {
+  const factory CreateDiscussionFormState({
     @Default('') String name,
-    AccessRecipient? memberAddFieldValue,
-    @Default([]) List<AccessRecipient> members,
-    AccessRecipient? adminAddFieldValue,
-    @Default([]) List<AccessRecipient> admins,
-  }) = _CreateDiscussionGroupFormState;
-  const CreateDiscussionGroupFormState._();
-
-  bool get canSubmit => members.isNotEmpty;
+    @Default([]) List<Contact> members,
+    @Default([]) List<Contact> admins,
+  }) = _CreateDiscussionFormState;
+  const CreateDiscussionFormState._();
+
+  bool get canSubmit => members.isNotEmpty && name.isNotEmpty;
+
+  bool get canGoNext =>
+      members.isNotEmpty &&
+      members.none((member) => PublicKey(member.publicKey).isValid == false);
+
+  List<Contact> get membersList => members;
 }
 
-class CreateDiscussionGroupFormNotifier
-    extends AutoDisposeNotifier<CreateDiscussionGroupFormState> {
-  CreateDiscussionGroupFormNotifier();
+class CreateDiscussionFormNotifier
+    extends AutoDisposeNotifier<CreateDiscussionFormState> {
+  CreateDiscussionFormNotifier();
 
   @override
-  CreateDiscussionGroupFormState build() =>
-      const CreateDiscussionGroupFormState();
-
-  void setMemberAddFieldValue(AddPublicKeyTextFieldValue fieldValue) {
-    state = state.copyWith(memberAddFieldValue: fieldValue.toAccessRecipient);
-  }
+  CreateDiscussionFormState build() => const CreateDiscussionFormState();
 
-  void addMember(AccessRecipient member) {
+  void addMember(Contact member) {
```

Comment:
Do you plan to remove completely `AccessRecipient` class ?

Reply:
I will check if we use it somewhere else, if not I will delete it.

Reply:
We are using it in NFT classes and in the add discussion sheet. So I am keeping it.